praw/models/reddit/subreddit.py

```diff
@@ -873,6 +877,120 @@ def submit(
 
         return self._reddit.post(API_PATH["submit"], data=data)
 
+    def submit_gallery(
+        self,
+        title,
+        images,
+        flair_id=None,
+        flair_text=None,
+        resubmit=True,
+        send_replies=True,
+        nsfw=False,
+        spoiler=False,
+        collection_id=None,
+        discussion_type=None,
+    ):
+        """Add an image gallery submission to the subreddit.
+
+        :param title: The title of the submission.
+        :param images: The images to post in dict with the following structure:
+            ``{"image_path": "path", "caption": "caption", "outbound_url": "url"}``,
+            only ``"image_path"`` is required.
+        :param collection_id: The UUID of a :class:`.Collection` to add the
+            newly-submitted post to.
+        :param flair_id: The flair template to select (default: None).
+        :param flair_text: If the template's ``flair_text_editable`` value is
+            True, this value will set a custom text (default: None).
+        :param resubmit: When False, an error will occur if the URL has already
+            been submitted (default: True).
+        :param send_replies: When True, messages will be sent to the submission
+            author when comments are made to the submission (default: True).
+        :param nsfw: Whether or not the submission should be marked NSFW
+            (default: False).
+        :param spoiler: Whether or not the submission should be marked as
+            a spoiler (default: False).
+        :param discussion_type: Set to ``CHAT`` to enable live discussion
+            instead of traditional comments (default: None).
+        :returns: A :class:`.Submission` object for the newly created submission.
+
+        If ``image_path`` in ``images`` refers to a file that is not an image, PRAW will
+        raise a :class:`.ClientException`.
+
+        For example to submit an image gallery to ``r/reddit_api_test`` do:
+
+        .. code-block:: python
+
+            title = "My favorite pictures"
+            image = "/path/to/image.png"
+            image2 = "/path/to/image2.png"
+            image3 = "/path/to/image3.png"
+            images = [
+               {
+                   "image_path": image
+               },
+               {
+                   "image_path": image2,
+                   "caption": "Image caption 2",
+               },
+               {
+                   "image_path": image3,
+                   "caption": "Image caption 3",
+                   "outbound_url": "https://example.com/link3"
+               }
+            ]
+            reddit.subreddit("reddit_api_test").submit_gallery(title, images)
+
+        .. seealso ::
+
+           * :meth:`.submit` to submit url posts and selftexts
+           * :meth:`.submit_image`. to submit single images
+           * :meth:`.submit_video`. to submit videos and videogifs
```

Comment:
Should we add `submit_poll` as well?

Reply:
Maybe? I didn't add it because it was in the scope of the PR. Though, I don't think it fits here since these are all ways to submit media.

Reply:
Good point, we can keep it out of this PR if you think that's better.